packages/flutter/lib/src/widgets/routes.dart

```diff
@@ -592,12 +595,18 @@ class _ModalScopeState<T> extends State<_ModalScope<T>> {
     if (widget.route.secondaryAnimation != null)
       animations.add(widget.route.secondaryAnimation);
     _listenable = Listenable.merge(animations);
+    if (widget.route.isCurrent) {
+      widget.route.navigator.focusScopeNode.setFirstFocus(focusScopeNode);
```

Comment:
According to https://docs.flutter.io/flutter/widgets/Route/navigator.html, a route's navigator can be null (here and below).

Reply:
It can only be null if it hasn't been pushed yet, or if it has been disposed.

In this case `isCurrent` checks that the navigator isn't null, and in the case of `didPush` below, by definition it has been pushed, so it's probably OK as-is.  I'm happy to still add the ?'s if you think it would be better to do it defensively against a future change, but they're not needed as-is.